clients/geth/specular/rollup/services/indexer/indexer.go

```diff
@@ -0,0 +1,72 @@
+package indexer
+
+import (
+	bind "github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/specularl2/specular/clients/geth/specular/proof"
+	"github.com/specularl2/specular/clients/geth/specular/rollup/services"
+)
+
+func RegisterService(stack *node.Node, eth services.Backend, proofBackend proof.Backend, cfg *services.Config, auth *bind.TransactOpts) {
+	indexer, err := New(eth, proofBackend, cfg, auth)
+	if err != nil {
+		log.Crit("Failed to register the Rollup service", "err", err)
+	}
+	stack.RegisterLifecycle(indexer)
+	// stack.RegisterAPIs(indexer.APIs())
+	log.Info("Indexer registered")
+}
+
+type Indexer struct {
```

Comment:
can we just fold this functionality into `validator.go` instead of having a separate class? just add some feature flagging to turn off assertion creation / challenging (just dont start the loop i guess); and then in service.go just switch on `cfg.Node` with a common case for `services.{NODE_VALIDATOR, NODE_INDEXER}`

i think that would improve maintainability / get rid of some common code

Reply:
we can list it as a separate task